File: 713. Subarray Product Less Than K/Solution.java
class Solution {
    public int numSubarrayProductLessThanK(int[] nums, int k) {
        if(k<=1) return 0;

        int count = 0;
        int right = 0, left = 0;

        int n = nums.length;
        int prod = 1;

        while(right < n){
            prod *= nums[right];

            while(left <= right && prod>=k){
                prod /= nums[left++];
            }

            count += right - left + 1;
            right++;            
        }

        return count;
    }
}

File: 713. Subarray Product Less Than K/Solution.py
class Solution:
    def numSubarrayProductLessThanK(self, nums: List[int], k: int) -> int:
        if k <= 1 : return 0

        left = right = count = 0
        prod = 1

        while right < len(nums):
            prod *= nums[right]

            while left <= right and prod >= k:
                prod //= nums[left]
                left += 1

            count += right - left + 1
            right += 1

        return count